### PR TITLE
Jacoco worky

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2082,7 +2095,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-report-plugin</artifactId>
-        <version>2.22.2</version>
+        <version>2.18</version>
```
Comment (kwatters) [inline pom.xml]: latest version is 3.2.2... any reason we can't go to the latest here?  (2.18 is from 2014..  nearly 10 years old at this point.)

Comment (supertick) [reply pom.xml]: after spending several hours on it and going through a variety of rabbit-holes "No", not if we want jacoco 0.8.11 (latest) to work

**Comment by supertick** [comment]
|Overall Project|8.05%|:x:|
|:-|:-|:-:|

> There is no coverage information present for the Files changed

**Comment by supertick** [comment]
|Overall Project|8.11%|:x:|
|:-|:-|:-:|

> There is no coverage information present for the Files changed

**Comment by supertick** [comment]
|Overall Project|17.43%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|56.65%|:green_apple:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.61%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|43.92%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|33.83%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|22.48%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|20.46%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|15.38%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|11.71%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|10.62%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/0b6496aa644b1dc291dfc01425333b9882f17ab2/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Comment by supertick** [comment]
|Overall Project|8.11%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.31%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|29.27%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|0%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|0%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|0%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|0%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|0%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|0%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|0%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/47133b98989f48607650bdf41106a4f75999b4ac/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Comment by supertick** [comment]
|Overall Project|17.43%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|56.65%|:green_apple:|
|[Outbox.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fframework%2FOutbox.java)|52.01%|:green_apple:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.61%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|43.92%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|33.83%|:green_apple:|
|[Util.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fimage%2FUtil.java)|32.29%|:green_apple:|
|[Vertx.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FVertx.java)|31.46%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|22.48%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|20.46%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|15.38%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|11.71%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|10.62%|:green_apple:|
|[Mail.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FMail.java)|0%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/c254391732c3e3e8bfdd27396c7b5aa992e6e75f/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Comment by supertick** [comment]
|Overall Project|17.6%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|56.65%|:green_apple:|
|[Outbox.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fframework%2FOutbox.java)|52.01%|:green_apple:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.61%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|43.92%|:green_apple:|
|[MapperSimple.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fmath%2FMapperSimple.java)|34.02%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|33.96%|:green_apple:|
|[Util.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fimage%2FUtil.java)|32.29%|:green_apple:|
|[Vertx.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FVertx.java)|31.46%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|22.48%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|20.46%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|15.38%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|11.71%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|10.62%|:green_apple:|
|[Mail.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FMail.java)|0%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/602cca803f52b7ab329ff4096cd41446a1ff32c5/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Comment by supertick** [comment]
|Overall Project|18.63%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|56.65%|:green_apple:|
|[Outbox.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fframework%2FOutbox.java)|52.01%|:green_apple:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.61%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|43.92%|:green_apple:|
|[MapperSimple.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fmath%2FMapperSimple.java)|34.02%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|33.96%|:green_apple:|
|[Util.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fimage%2FUtil.java)|32.29%|:green_apple:|
|[Vertx.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FVertx.java)|31.46%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|22.48%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|20.46%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|15.38%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|11.71%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|10.62%|:green_apple:|
|[Mail.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FMail.java)|0%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/09e9567ec8f6af2772b4e70c78acd467cdf2d8a1/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Comment by supertick** [comment]
|Overall Project|18.81%|:x:|
|:-|:-|:-:|

|File|Coverage||
|:-|:-|:-:|
|[NovelAI.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNovelAI.java)|56.65%|:green_apple:|
|[Outbox.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fframework%2FOutbox.java)|52.01%|:green_apple:|
|[Python.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPython.java)|45.61%|:green_apple:|
|[Py4j.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPy4j.java)|43.92%|:green_apple:|
|[MapperSimple.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fmath%2FMapperSimple.java)|34.02%|:green_apple:|
|[Runtime.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FRuntime.java)|33.96%|:green_apple:|
|[Util.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fimage%2FUtil.java)|32.29%|:green_apple:|
|[Vertx.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FVertx.java)|31.46%|:green_apple:|
|[NeoPixel.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FNeoPixel.java)|22.48%|:green_apple:|
|[LocalSpeech.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FLocalSpeech.java)|20.46%|:green_apple:|
|[_TemplateService.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2F_TemplateService.java)|15.38%|:green_apple:|
|[MotorControl.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2Finterfaces%2FMotorControl.java)|15.38%|:green_apple:|
|[Pir.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FPir.java)|11.71%|:green_apple:|
|[ServoMixer.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fservice%2FServoMixer.java)|10.62%|:green_apple:|
|[Mail.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FMail.java)|0%|:green_apple:|
|[SslUtil.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fnet%2FSslUtil.java)|0%|:green_apple:|
|[PredicateEvent.java](https://github.com/MyRobotLab/myrobotlab/blob/a535fa924416423e3ebf2b141f1f700ef83786f3/src%2Fmain%2Fjava%2Forg%2Fmyrobotlab%2Fprogramab%2FPredicateEvent.java)|0%|:green_apple:|

**Pull request closed without merging**